python/addBTV.py

```diff
@@ -299,6 +299,91 @@ def add_BTV(process, runOnMC=False, onlyAK4=False, onlyAK8=False, keepInputs=['D
                       doc="DeepCSV light btag discriminator",
                       precision=10),
     )
+
+    FatJetVars = cms.PSet(
+        uncorrpt=Var("?availableJECSets().size()>0?correctedJet('Uncorrected').pt():pt()",
+                     float,
+                     doc="Uncorrected pT",
+                     precision=10),
+        residual=Var("?availableJECSets().size()>0 ? pt()/correctedJet('L3Absolute').pt() : 1. ",
+                     float,
+                     doc="residual",
+                     precision=10),
+        jes=Var("?availableJECSets().size()>0 ? pt()/correctedJet('Uncorrected').pt() : 1.",
+                float,
+                doc="jes (jet substructure)",
+            precision=10),
+        CombIVF=Var("bDiscriminator('pfCombinedInclusiveSecondaryVertexV2BJetTags')",
+                float,
+                doc="combinedIVF",
+                precision=10),
+        DeepCSVBDisc=Var("bDiscriminator('pfDeepCSVJetTags:probb')+bDiscriminator('pfDeepCSVJetTags:probbb')",
+            float,
+            doc="DeepCSVBDisc",
+            precision=10),
+        btagDeepB_c=Var("bDiscriminator('pfDeepCSVJetTags:probc')",
+                        float,
+                        doc="DeepCSV c tag discriminator",
+                        precision=10),
+        DeepCSVCvsLDisc=Var("? (bDiscriminator('pfDeepCSVJetTags:probc'))!=-1 ? bDiscriminator('pfDeepCSVJetTags:probc')/(bDiscriminator('pfDeepCSVJetTags:probc')+bDiscriminator('pfDeepCSVJetTags:probudsg')) : -1",
+            float,
+            doc="DeepCSV C vs L discriminator",
+            precision=10),
+        DeepCSVCvsBDisc=Var("? (bDiscriminator('pfDeepCSVJetTags:probc'))!=-1 ? bDiscriminator('pfDeepCSVJetTags:probc')/(bDiscriminator('pfDeepCSVJetTags:probc')+bDiscriminator('pfDeepCSVJetTags:probb')+bDiscriminator('pfDeepCSVJetTags:probbb')) : -1",
+            float,
+            doc="DeepCSV C vs B discriminator",
+            precision=10),
```

Comment:
[Implementation itself, technical details & storage]
We might want to synchronize this check for invalid values with the statement in NanoAOD, example here: https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/NanoAOD/python/jetsAK4_Puppi_cff.py#L86-L87 —> switch from `!=-1` to `>=0`. Although that should not make that big of a difference, probably it has some effects related to floating point precision.